MapleServer2/PacketHandlers/Game/ItemEnchantHandler.cs

```diff
@@ -137,25 +149,24 @@ private static ItemEnchant GetEnchantInfo(Item item)
             itemEnchantStats.Ingredients.Add(ingredient);
         }
 
-        for (int i = 0; i < statValueScriptResult.Tuple.Length; i += 2)
-        {
-            if (statValueScriptResult.Tuple[i].Number == 0)
-            {
-                continue;
-            }
-
-            StatAttribute attribute = (StatAttribute) statValueScriptResult.Tuple[i].Number;
-            float boostRate = (float) statValueScriptResult.Tuple[i + 1].Number;
-            itemEnchantStats.Stats[attribute] = new(attribute, 0, boostRate);
-        }
+        itemEnchantStats.Stats = EnchantHelper.GetEnchantStats(item.EnchantLevel + 1, item.Type, item.Level);
         return itemEnchantStats;
     }
 
     private static void HandleUpdateCharges(GameSession session, PacketReader packet)
     {
         int chargeCount = packet.ReadInt();
         ItemEnchant itemEnchant = session.Player.ItemEnchant;
-        Item item = session.Player.Inventory.GetByUid(itemEnchant.ItemUid);
+        Item item = null;
+
+        if (session.Player.Inventory.HasItem(itemEnchant.ItemUid))
+        {
+            item = session.Player.Inventory.GetByUid(itemEnchant.ItemUid);
+        }
+        else if (session.Player.ItemIsEquipped(itemEnchant.ItemUid))
+        {
+            item = session.Player.GetEquippedItem(itemEnchant.ItemUid);
+        }
```

Comment:
Maybe add an function to Inventory get the item from inventory or the equipped inventory? Maybe could be useful later on too